packages/ag-charts-community/src/chart/data/processors.ts

```diff
@@ -307,16 +310,16 @@ const animationValidationProcessKey = (
     let validation = ANIMATION_VALIDATION_UNIQUE_KEYS | ANIMATION_VALIDATION_ORDERED_KEYS;
 
     if (def.valueType === 'category') {
-        if (!(keyValues.length === count)) validation &= ~ANIMATION_VALIDATION_UNIQUE_KEYS;
+        if (keyValues.length !== count) validation &= ~ANIMATION_VALIDATION_UNIQUE_KEYS;
 
         return validation;
     }
 
     let lastValue = column[0];
     for (let d = 1; validation !== 0 && d < column.length; d++) {
         const keyValue = column[d];
-        if (!(lastValue <= keyValue)) validation &= ~ANIMATION_VALIDATION_ORDERED_KEYS;
-        if (!(lastValue !== keyValue)) validation &= ~ANIMATION_VALIDATION_UNIQUE_KEYS;
+        if (lastValue > keyValue) validation &= ~ANIMATION_VALIDATION_ORDERED_KEYS;
```

Comment:
Not quite - the old version handled NaNs

Reply:
We should make that a bit more explicit here perhaps, can we handle the `NaN` case explicitly?

Reply:
Only thing to remember is it might have date objects in it. Maybe just call .valueOf to get a numeric value, then use Number.isFinite and the normal comparison operators

Reply:
Reverted for now, we'll follow-up tomorrow.